docs: Update registry index file and main readme (#272)
elastic/terraform-provider-ec: docs/index.md

## Releases

Interested in the provider's latest features, or want to make sure you're up to date? [Check out the provider changelog](https://github.com/elastic/terraform-provider-ec/blob/master/CHANGELOG.md).

## Authentication

The Elastic Cloud Terraform provider offers two methods of authentication against the remote API: `apikey` or a combination of `username` and `password`. Depending on the environment, you may choose one over the other. The Public API of Elasticsearch Service (ESS) is the default `endpoint` that the provider will target.

Elasticsearch Service (ESS) only supports `apikey`. Elastic Cloud Enterprise (ECE) supports `apikey` or a combination of `username` and `password`.

!> **Warning:** Hard-coding credentials into a Terraform configuration is not recommended, and risks secret leakage should this file ever be committed to a public version control system.

### API key authentication (recommended)

API keys are the recommended authentication method. They can be used to authenticate against Elasticsearch Service or Elastic Cloud Enterprise.

#### Generating an Elasticsearch Service (ESS) API Key

To generate an API key, follow these steps:

>   1. Navigate to <https://cloud.elastic.co/login> with your browser

Reviewer: ```suggestion
  1. Open you browser and navigate to <https://cloud.elastic.co/login>.
```

>   2. Log in with your Email and Password.

Reviewer: ```suggestion
  2. Log in with your email and password.
```

  3. Click on [Elasticsearch Service](https://cloud.elastic.co/deployments).

>   4. Navigate to [Featuress > API Keys](https://cloud.elastic.co/deployment-features/keys) and click on **Generate API Key**.

Reviewer: ```suggestion
  4. Navigate to [Features > API Keys](https://cloud.elastic.co/deployment-features/keys) and click on **Generate API Key**.
```

>   5. Chose a name for your API key.

Reviewer: ```suggestion
  5. Choose a name for your API key.
```

>   6. Persist your API key somewhere safe.

Reviewer: ```suggestion
  6. Save your API key somewhere.
```

#### Using your API Key on the Elastic Cloud terraform provider

> After you've generated your API Key, you can make it available to the Terraform provider by exporting it as the environment variable `EC_API_KEY` environment variable (recommended) or hardcoded in the provider `.tf` configuration file (supported but not recommended).

Reviewer: ```suggestion
After you've generated your API Key, you can make it available to the Terraform provider by exporting it as the environment variable `EC_API_KEY` (recommended), or hardcoded in the provider `.tf` configuration file (supported but not recommended).
```
